src/main/java/org/mockito/internal/invocation/ArgumentsProcessor.java

```diff
@@ -9,15 +9,25 @@
 import org.mockito.internal.matchers.Equals;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * by Szczepan Faber, created at: 3/31/12
  */
 public class ArgumentsProcessor {
+    // drops hidden synthetic parameters (last continuation parameter from Kotlin suspending functions)
```

Comment:
Again, thanks for the code comments! Very useful in long term maintenance of the project.